### PR TITLE
ST: Improve logs of system tests to make it more readable

### PR DESCRIPTION
**Pull request by Frawless**
### Type of change

- Refactoring

### Description

This PR change level of a several outputs and overall change logging in all system tests to make the output more readable and clear in case of debugging.

[info log output](https://gist.github.com/Frawless/c5f682973e7197d22ef1610aeb3a5af6)
[debug log output](https://gist.github.com/Frawless/bf53ea6d157c1a1632ae9ffdb089b09c)

### Checklist

- [ ] Make sure all tests pass

### CODE REVIEW

File: systemtest/src/main/java/io/strimzi/systemtest/resources/operator/SetupClusterOperator.java [hunk 1]
```diff
@@ -190,15 +191,16 @@ public SetupClusterOperatorBuilder defaultInstallation() {
      * Don't use this method in tests, where specific configuration of CO is needed.
      */
     public SetupClusterOperator runInstallation() {
-        LOGGER.info("Cluster operator installation configuration:\n{}", this::toString);
+        LOGGER.info("Cluster operator installation configuration:\n{}", this::prettyPrint);
+        LOGGER.debug("Cluster operator installation configuration:\n{}", this::toString);
```
Comment (im-konge) [inline systemtest/src/main/java/io/strimzi/systemtest/resources/operator/SetupClusterOperator.java:195]: Is this needed here? We have almost the same for the info log level

**Review by im-konge — APPROVED** [review]
LGTM, just one nit

File: systemtest/src/main/java/io/strimzi/systemtest/resources/operator/SetupClusterOperator.java (hunk 1 above)
Comment (Frawless) [reply systemtest/src/main/java/io/strimzi/systemtest/resources/operator/SetupClusterOperator.java:195]: Discussed offline

**Review by see-quick — APPROVED** [review]

**Comment by see-quick** [comment]
I think you should also change the `log4j2.properties` file. In case someone wants to see the `TRACE` level of the current situation it wouldn't be possible. I guess you have to change `rootLogger.level` setting to `TRACE` instead of `DEBUG`.

**Comment by Frawless** [comment]
Wouldn't be better to set it based on `STRIMZI_TEST_LOG_LEVEL` ?

**Comment by scholzj** [comment]
Is it intentional to have TRACE level set as the default enabled log level? That does not sound like logging done well to me.

**Comment by Frawless** [comment]
The default for a console logger is INFO and for the file is DEBUG. The latest change just allows us to set TRACE via `STRIMZI_TEST_LOG_LEVEL`. At least It worked like that for me yesterday when I tested it.

**Comment by scholzj** [comment]
Sorry, how do these two lines _keep INFO as default and allow to set TRACE through env var?

```
rootLogger.level = TRACE
rootLogger.appenderRef.console.level = ${env:STRIMZI_TEST_LOG_LEVEL:-TRACE}
```

They set the TRACE as the default log level - or that is at least the way I read it:
* Default root log level is TRACE
* When `STRIMZI_TEST_LOG_LEVEL` is not set, default to TRACE

**Comment by Frawless** [comment]
Ah sorry, I thought I reverted my change for default value for the env var. Good catch, going to revert it

**Comment by scholzj** [comment]
> Ah sorry, I thought I reverted my change for default value for the env var. Good catch, going to revert it

What about the `rootLogger.level`? That seems to be still TRACE.

**Comment by Frawless** [comment]
ye, that is as I mentioned above. It is just a maximum allowed log level

**Comment by scholzj** [comment]
Why would that be highest allowed level? Any docs links to point to? That sounds to me like the very default level. And then you override it for the console appender and for the rolling appender. But if you add a new appender, it would by default use TRACE.

File: systemtest/src/main/java/io/strimzi/systemtest/logs/LogCollector.java [hunk 2]
```diff
@@ -212,13 +213,13 @@ private void collectReplicaSets(String namespace) {
 
     private void collectStrimzi(String namespace) {
         LOGGER.info("Collecting Strimzi in Namespace {}", namespace);
-        String crData = cmdKubeClient(namespace).exec(false, false, "get", "strimzi", "-o", "yaml", "-n", namespaceFile.getName()).out();
+        String crData = cmdKubeClient(namespace).exec(false, StandardLevel.DEBUG, "get", "strimzi", "-o", "yaml", "-n", namespaceFile.getName()).out();
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/logs/LogCollector.java:216]: Are you sure StandardLevel.DEBUG is the right thing to use? Javadoc for this package suggests it is _Internal interfaces and classes to be used by authors of logging implementations or for internal use by API classes._ I would expect this https://logging.apache.org/log4j/2.x/log4j-api/apidocs/index.html should be used instead?

File: systemtest/src/main/java/io/strimzi/systemtest/logs/LogCollector.java [hunk 3]
```diff
@@ -212,13 +213,13 @@ private void collectReplicaSets(String namespace) {
...
         writeFile(namespaceFile + "/strimzi-custom-resources.log", crData);
     }
 
     private void collectClusterInfo(String namespace) {
         LOGGER.info("Collecting cluster status");
-        String nodes = cmdKubeClient(namespace).exec(false, false, "describe", "nodes").out();
+        String nodes = cmdKubeClient(namespace).exec(false, StandardLevel.DEBUG, "describe", "nodes").out();
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/logs/LogCollector.java:222]: Same as above.

File: systemtest/src/main/java/io/strimzi/systemtest/resources/operator/SetupClusterOperator.java [hunk 4]
```diff
@@ -406,7 +408,7 @@ public void prepareEnvForOperator(ExtensionContext extensionContext, String clie
             if (kubeClient().getNamespace(Environment.STRIMZI_ORG) != null) {
                 for (String namespace : namespaces) {
                     LOGGER.debug("Setting group policy for Openshift registry in namespace: " + namespace);
-                    Exec.exec(null, Arrays.asList("oc", "policy", "add-role-to-group", "system:image-puller", "system:serviceaccounts:" + namespace, "-n", Environment.STRIMZI_ORG), 0, false, false);
+                    Exec.exec(null, Arrays.asList("oc", "policy", "add-role-to-group", "system:image-puller", "system:serviceaccounts:" + namespace, "-n", Environment.STRIMZI_ORG), 0, StandardLevel.DEBUG, false);
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/resources/operator/SetupClusterOperator.java:411]: Same as above.

File: systemtest/src/main/java/io/strimzi/systemtest/resources/operator/specific/OlmResource.java [hunk 5]
```diff
@@ -287,7 +288,7 @@ private static void waitFor(String deploymentName, String namespace, int replica
     }
 
     private static Map<String, JsonObject> parseExamplesFromCsv(String csvName, String namespace) {
-        String csvString = ResourceManager.cmdKubeClient().exec(true, false, "get", "csv", csvName, "-o", "json", "-n", namespace).out();
+        String csvString = ResourceManager.cmdKubeClient().exec(true, StandardLevel.DEBUG, "get", "csv", csvName, "-o", "json", "-n", namespace).out();
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/resources/operator/specific/OlmResource.java:291]: Same as above.

File: systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java [hunk 6]
```diff
@@ -250,7 +251,7 @@ public static void checkLogForJSONFormat(String namespaceName, Map<String, Strin
         TestUtils.waitFor("for JSON log in " + pods, Constants.GLOBAL_POLL_INTERVAL_MEDIUM, Constants.GLOBAL_TIMEOUT, () -> {
             boolean isJSON = false;
             for (String podName : pods.keySet()) {
-                String log = cmdKubeClient().namespace(namespaceName).execInCurrentNamespace(false, "logs", podName, "-c", containerName, tail).out();
+                String log = cmdKubeClient().namespace(namespaceName).execInCurrentNamespace(StandardLevel.TRACE, "logs", podName, "-c", containerName, tail).out();
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java:254]: Same as above ...

File: systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaConnectUtils.java [hunk 7]
```diff
@@ -72,7 +73,7 @@ public static void waitUntilKafkaConnectRestApiIsAvailable(String podNamePrefix)
     public static void waitForMessagesInKafkaConnectFileSink(String namespaceName, String kafkaConnectPodName, String sinkFileName, String message) {
         LOGGER.info("Waiting for messages in file sink on {}", kafkaConnectPodName);
         TestUtils.waitFor("messages in file sink", Constants.GLOBAL_POLL_INTERVAL, Constants.TIMEOUT_FOR_SEND_RECEIVE_MSG,
-            () -> cmdKubeClient(namespaceName).execInPod(false, kafkaConnectPodName, "/bin/bash", "-c", "cat " + sinkFileName).out().contains(message));
+            () -> cmdKubeClient(namespaceName).execInPod(StandardLevel.TRACE, kafkaConnectPodName, "/bin/bash", "-c", "cat " + sinkFileName).out().contains(message));
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaConnectUtils.java:76]: Same as above ...

File: systemtest/src/main/java/io/strimzi/systemtest/utils/specific/CruiseControlUtils.java [hunk 8]
```diff
@@ -62,7 +63,7 @@ public static String callApi(String namespaceName, SupportedHttpMethods method,
                 + " --user admin:$(cat /opt/cruise-control/api-auth-config/cruise-control.apiAdminPassword) ";
         }
 
-        return cmdKubeClient(namespaceName).execInPodContainer(false, ccPodName, CONTAINER_NAME, "/bin/bash", "-c",
+        return cmdKubeClient(namespaceName).execInPodContainer(StandardLevel.DEBUG, ccPodName, CONTAINER_NAME, "/bin/bash", "-c",
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/specific/CruiseControlUtils.java:66]: Same as above ...

File: systemtest/src/main/java/io/strimzi/systemtest/utils/specific/CruiseControlUtils.java [hunk 9]
```diff
@@ -71,7 +72,7 @@ public static String callApi(String namespaceName, SupportedHttpMethods method,
     public static String callApi(String namespaceName, SupportedHttpMethods method, String endpoint) {
         String ccPodName = PodUtils.getFirstPodNameContaining(namespaceName, CONTAINER_NAME);
 
-        return cmdKubeClient(namespaceName).execInPodContainer(false, ccPodName, CONTAINER_NAME, "/bin/bash", "-c",
+        return cmdKubeClient(namespaceName).execInPodContainer(StandardLevel.DEBUG, ccPodName, CONTAINER_NAME, "/bin/bash", "-c",
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/specific/CruiseControlUtils.java:75]: Same as above ...

File: systemtest/src/main/java/io/strimzi/systemtest/utils/specific/KeycloakUtils.java [hunk 10]
```diff
@@ -35,7 +36,7 @@ public static void deployKeycloak(String namespace) {
 
         // This is needed because from time to time the first try fails on Azure
         TestUtils.waitFor("Keycloak instance readiness", Constants.KEYCLOAK_DEPLOYMENT_POLL, Constants.KEYCLOAK_DEPLOYMENT_TIMEOUT, () -> {
-            ExecResult result = Exec.exec(true, "/bin/bash", PATH_TO_KEYCLOAK_PREPARE_SCRIPT, namespace, getValidKeycloakVersion());
+            ExecResult result = Exec.exec(StandardLevel.INFO, "/bin/bash", PATH_TO_KEYCLOAK_PREPARE_SCRIPT, namespace, getValidKeycloakVersion());
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/specific/KeycloakUtils.java:39]: Same as above ...

File: systemtest/src/main/java/io/strimzi/systemtest/utils/specific/KeycloakUtils.java [hunk 11]
```diff
@@ -49,7 +50,7 @@ public static void deployKeycloak(String namespace) {
 
     public static void deleteKeycloak(String namespace) {
         LOGGER.info("Teardown Keycloak in namespace: {}", namespace);
-        Exec.exec(true, "/bin/bash", PATH_TO_KEYCLOAK_TEARDOWN_SCRIPT, namespace, getValidKeycloakVersion());
+        Exec.exec(StandardLevel.INFO, "/bin/bash", PATH_TO_KEYCLOAK_TEARDOWN_SCRIPT, namespace, getValidKeycloakVersion());
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/specific/KeycloakUtils.java:53]: Same as above ...

File: systemtest/src/test/java/io/strimzi/systemtest/backup/ColdBackupScriptIsolatedST.java [hunk 12]
```diff
@@ -70,7 +71,7 @@ void backupAndRestore(ExtensionContext context) {
         String[] backupCommand = new String[] {
             USER_PATH + "/../tools/cold-backup/run.sh", "backup", "-n", INFRA_NAMESPACE, "-c", clusterName, "-t", backupFilePath, "-y"
         };
-        Exec.exec(true, backupCommand);
+        Exec.exec(StandardLevel.INFO, backupCommand);
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/backup/ColdBackupScriptIsolatedST.java:74]: Same as above ... and in all the cases below ...

**Review by scholzj — APPROVED** [review]

**Pull request merged**